src/functions/commerce/provider.php

```diff
@@ -48,4 +48,36 @@ function tribe_tickets_commerce_is_test_mode() {
 	 * @todo This method likely should be focused on paypal only usage as it's gateway specific conditional.
 	 */
 	return \TEC\Tickets\Commerce\Gateways\PayPal\Gateway::is_test_mode();
+}
+
+/**
+ * Determine whether the legacy TribeCommerce should be shown or not.
+ *
+ * @since TBD
+ *
+ * @return boolean
+ */
+function tec_tribe_commerce_is_available() {
+
+	if ( defined( 'TEC_TRIBE_COMMERCE_AVAILABLE' ) ) {
+		return (bool) TEC_TRIBE_COMMERCE_AVAILABLE;
+	}
+
+	$env_var = getenv( 'TEC_TRIBE_COMMERCE_AVAILABLE' );
+
+	if ( false !== $env_var ) {
+		return (bool) $env_var;
+	}
+
+	// Available if PayPal was completely setup previously.
+	$available = tribe( 'tickets.commerce.paypal.handler.ipn' )->get_config_status( 'slug' ) === 'complete';
```

Comment:
This class might also be a problem if called too early, so instead of fatalling we should check if it's available to consume and if not use `_doing_it_wrong()`

Reply:
Also yoda